include/mbedtls/cipher.h

```diff
@@ -479,7 +492,7 @@ static inline size_t mbedtls_cipher_info_get_key_bitlen(
     if (info == NULL) {
         return 0;
     } else {
-        return info->MBEDTLS_PRIVATE(key_bitlen);
+        return info->MBEDTLS_PRIVATE(key_bitlen) << MBEDTLS_KEY_BITLEN_SHIFT;
```

Comment:
The largest value we could see here would fit into 10 bits (because it's a 4-bit bitfield shifted 6 places), so there is no danger of over-flowing the unsigned int. The cast is harmless but I don't think it adds anything.

Reply:
Some compilers (MSVC, recent clang with a flag that we don't enable by default but some users like, IAR) warn about (some) conversions of integers to a smaller type. In theory `size_t` could be smaller than `unsigned`, but in practice it isn't (and if it is we probably have other problems due to unexpected promotions of `size_t` values to signed `int`). So casting from `unsigned` to `size_t` is not necessary (but from `size_t` to `unsigned` would be).

Reply:
> The largest value we could see here would fit into 10 bits (because it's a 4-bit bitfield shifted 6 places), so there is no danger of over-flowing the unsigned int. The cast you suggest is fine IMO but I don't think it adds anything. So I won't change this unless you or @tom-cosgrove-arm recommend this?

Implicit sign conversion breaks compilation in other projects that enforce warnings; e.g.: this happens when building Mongoose in MacOS:
```c
/usr/local/Cellar/mbedtls/3.5.0/include/mbedtls/cipher.h:495:50: error: implicit conversion changes signedness: 'int' to 'size_t' (aka 'unsigned long') [-Werror,-Wsign-conversion]
            return info->MBEDTLS_PRIVATE(key_bitlen) << MBEDTLS_KEY_BITLEN_SHIFT;
            ~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
```
The cast suggested by @yanrayw does add something.